lib/auth/bot.go

```diff
@@ -189,9 +191,26 @@ func (a *ServerWithRoles) updateBotInstance(ctx context.Context, req *certReques
 		return nil
 	}
 
+	// TODO(nklaassen): consider recording both public keys once they are
+	// actually separated.
+	var publicKeyPEM []byte
+	if req.tlsPublicKey != nil {
+		publicKeyPEM = req.tlsPublicKey
+	} else {
+		// At least one key will be set, this is validated by [req.check].
+		cryptoPubKey, err := sshutils.CryptoPublicKey(req.sshPublicKey)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		publicKeyPEM, err = keys.MarshalPublicKey(cryptoPubKey)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
 	authRecord := &machineidv1pb.BotInstanceStatusAuthentication{
 		AuthenticatedAt: timestamppb.New(a.authServer.GetClock().Now()),
-		PublicKey:       req.publicKey,
+		PublicKey:       publicKeyPEM,
```

Comment:
it may look like I'm changing this from SSH authorized key format to PEM format, and I am, but noah has a PR in flight to do the same https://github.com/gravitational/teleport/pull/43845